src/main/scala/mesosphere/marathon/state/Migration.scala

```diff
@@ -244,6 +290,29 @@ class MigrationTo0_11(groupRepository: GroupRepository, appRepository: AppReposi
   }
 }
 
+class MigrationTo0_12(taskRepository: TaskRepository) {
+  private[this] val log = LoggerFactory.getLogger(getClass)
+
+  val entityStore = taskRepository.entityStore
+
+  def migrateTasks(): Future[Unit] = {
+    log.info("Start 0.12 migration")
+
+    def migrateKey(legacyKey: String): Future[Unit] = {
+      entityStore.fetch(legacyKey).flatMap {
+        case Some(taskState) => taskRepository.store(taskState.task).flatMap { _ =>
```

Comment:
Can you add a short comment why you are doing this (task:foo:foo:version -> task:foo:verison)